components/brave_today/rust/lib.rs

```diff
@@ -0,0 +1,121 @@
+        voca_rs::strip::strip_tags(&title.content)
+    } else {
+        String::new()
+    };
+    for feed_item_data in feed.entries {
+        // Check we can make a valid entry
+        if feed_item_data.links.is_empty()
+            || feed_item_data.published.is_none()
+            || (feed_item_data.title.is_none() && feed_item_data.summary.is_none())
+            || feed_item_data.published.is_none()
+        {
+            // Ignore if we can't use this entry
+            continue;
+        }
+        let mut image_url = String::new();
+        // Attempt to get an image by looking at a couple possible fields:
+        //  - media[].content[].url
+        //  - media[].thumbnails[].image.uri
+        for media_object in feed_item_data.media {
+            let mut largest_width = 0;
+            for content_item in media_object.content {
+                if let Some(content_item_url) = content_item.url {
+                    let this_width = content_item.width.unwrap_or(0);
+                    if this_width >= largest_width {
+                        image_url = content_item_url.to_string();
+                        largest_width = this_width;
+                    }
+                }
+            }
+            if !image_url.is_empty() {
+                continue;
+            }
+            largest_width = 0;
+            for content_item in media_object.thumbnails {
+                if !content_item.image.uri.is_empty() {
+                    let this_width = content_item.image.width.unwrap_or(0);
+                    if this_width >= largest_width {
+                        image_url = content_item.image.uri;
+                        largest_width = this_width;
+                    }
+                }
+            }
+        }
+
+        let mut summary: String = String::new();
+        if feed_item_data.summary.is_some() {
+            summary = feed_item_data.summary.unwrap().content;
+        } else if feed_item_data.content.is_some() {
+            summary = feed_item_data.content.unwrap().body.unwrap_or_else(String::new);
+        }
+        // If we still don't have an image, attempt to parse an html
+        // <img> element from the summary. This is not uncommon.
+        if image_url.is_empty() && !summary.is_empty() {
+            // This relies on the string being already html-decoded, which
+            // feed-rs (so far) does.
+            let optional_caps = IMAGE_REGEX.captures(&summary);
+            if let Some(caps) = optional_caps {
+                if let Some(capture) = caps.get(1) {
+                    image_url = String::from(capture.as_str());
+                }
+            }
+        }
+        let feed_item = ffi::FeedItem {
+            id: feed_item_data.id,
+            title: voca_rs::strip::strip_tags(
+                &(if feed_item_data.title.is_some() {
+                    feed_item_data.title.unwrap().content
+                } else {
+                    summary.clone()
+                }),
```

Comment:
Ugh. The api makes this tricky. You shouldn't need to `summary.clone()` just to reference it again, but I couldn't find a clearer way to express this.